### PR TITLE
proto: add method support setupintent

### PR DESCRIPTION
**Pull request by Daydaylw3**

### CODE REVIEW

File: proto/billing.proto [hunk 1]
```diff
@@ -7,15 +7,33 @@ option go_package = "appflow.ai/onegenerative-ai-billing-service/gen/billing";
 import "google/protobuf/empty.proto";
 
 service BillingService {
+  // Used for Setup Intent API
+  rpc CreateCheckoutSession (CreateCheckoutSessionRequest) returns (CreateCheckoutSessionResponse) {}
```
Comment (Chellhooo) [inline proto/billing.proto:11]: `mode` can be 'subscription ' or 'setup'

File: proto/billing.proto [hunk 2]
```diff
@@ -7,15 +7,33 @@ option go_package = "appflow.ai/onegenerative-ai-billing-service/gen/billing";
...
   rpc DeleteCard (DeleteCardRequest) returns (google.protobuf.Empty) {}
 
+  // Deprecated use setupintent to add payment method
   rpc ListCards (ListCardsRequest) returns (ListCardsResponse) {}
+
+  rpc ListPaymentMethods (ListPaymentMethodsRequest) returns (ListPaymentMethodsResponse) {}
+
+  rpc DeletePaymentMethod (DeletePaymentMethodRequest) returns (google.protobuf.Empty) {}
+}
+
```
Comment (Chellhooo) [inline proto/billing.proto:30]: need `createSubscription` and `getSubscription`(by customer id) api also

File: proto/billing.proto [hunk 3]
```diff
@@ -71,3 +89,24 @@ message ListCardsRequest {
...
+  repeated PaymentMethod payment_methods = 1;
+}
+
+message PaymentMethod {
+  string id = 1;        // payment method id
+  string type = 2;      // ["card"]
+  int64 created_at = 3; // up to ms
+
+  Card card = 4;        // occurs when type is "card", has no `id`
+}
```
Comment (Chellhooo) [inline proto/billing.proto:107]: frontend also need to know which is default payment method

File: proto/billing.proto (hunk 2 above)
Comment (Daydaylw3) [reply proto/billing.proto:30]: will later on

File: proto/billing.proto (hunk 3 above)
Comment (Daydaylw3) [reply proto/billing.proto:107]: this is in customer, will add future

File: proto/billing.proto [hunk 4]
```diff
@@ -7,15 +7,33 @@ option go_package = "appflow.ai/onegenerative-ai-billing-service/gen/billing";
...
+  // Deprecated use setupintent to add payment method
   rpc ListCards (ListCardsRequest) returns (ListCardsResponse) {}
+
+  rpc ListPaymentMethods (ListPaymentMethodsRequest) returns (ListPaymentMethodsResponse) {}
+
+  rpc DeletePaymentMethod (DeletePaymentMethodRequest) returns (google.protobuf.Empty) {}
+}
+
+message CreateCheckoutSessionRequest {
+  string customer_id = 1;
```
Comment (zzn01) [inline proto/billing.proto:32]: Does frontend need customer_id? @Chellhooo

Comment (Daydaylw3) [reply proto/billing.proto:32]: yes, flow is createUser => frontend get customer info => add payment method and provide customer_id

Comment (Daydaylw3) [reply proto/billing.proto]: ok

**Review by zzn01 — APPROVED** [review]

**Pull request merged**